### PR TITLE
Feat: Adding room,  update room question stack and adding team to room

### PR DESCRIPTION
**Pull request by AnataAria**

### CODE REVIEW

File: src/rooms/rooms.service.ts [hunk 1]
```diff
@@ -222,4 +240,84 @@ export class RoomsService {
...
+  async addAllTeamsToRoom(roomTeam: CreateScoreTeamDto) {
+    if (!roomTeam?.roomCode) return [null, 'Enter room code and try again'];
+    const roomEntity: Room | undefined = await this.roomRepository.findOne({
+      where: {
+        code: roomTeam.roomCode,
+      },
+    });
+    if (!roomEntity)
+      return [null, `Not found room with code ${roomTeam.roomCode}`];
+    if (roomTeam.teamIds.length == 0) return ['Adding team successfully', null];
```
Comment (CuonggTV) [inline src/rooms/rooms.service.ts:280]: Should return error teams is missing

File: src/rooms/rooms.service.ts [hunk 2]
```diff
@@ -222,4 +240,84 @@ export class RoomsService {
...
+    const teamList = await this.dataSource
+      .getRepository(Team)
+      .createQueryBuilder()
+      .select('*')
+      .where('id IN (:...ids)', { ids: roomTeam.teamIds })
+      .getMany();
```
Comment (CuonggTV) [inline src/rooms/rooms.service.ts:285]: Cannot get data. Try using getRawMany() instead

File: src/rooms/rooms.service.ts [hunk 3]
```diff
@@ -222,4 +240,84 @@ export class RoomsService {
...
+    for (const team of teamList) {
```
Comment (CuonggTV) [inline src/rooms/rooms.service.ts]: Check if teamList is empty.

File: src/rooms/rooms.service.ts [hunk 4]
```diff
@@ -222,4 +240,84 @@ export class RoomsService {
...
+        const scoreList: Score[] = [];
+        teamList.forEach((team) => {
+          const tempScore = new Score();
+          tempScore.room = roomEntity;
+          tempScore.penalty = 0;
+          tempScore.totalScore = 0;
+          tempScore.team = team;
+          scoreList.push(tempScore);
+        });
+        await manager.save(scoreList);
```
Comment (CuonggTV) [inline src/rooms/rooms.service.ts]: Database errror => This code can duplicate "team_id" with "room_id" in table scores

File: src/rooms/rooms.service.ts (hunk 1 above)
Comment (AnataAria) [reply src/rooms/rooms.service.ts:280]: No, u shouldn't do that, that is not an error, I add this to not make a unuse query

File: src/rooms/rooms.service.ts (hunk 3 above)
Comment (AnataAria) [reply src/rooms/rooms.service.ts]: I think not need this, but undefine is ok, let me do this

File: src/rooms/rooms.service.ts (hunk 4 above)
Comment (AnataAria) [reply src/rooms/rooms.service.ts]: I have handle it on catch

File: src/rooms/rooms.service.ts [hunk 5]
```diff
@@ -222,4 +240,89 @@ export class RoomsService {
...
+    const stackResult: QuestionStack | undefined =
+      await this.questionStackRepository.findOne({
+        where: {
+          id: stackId,
+          status: QuestionStackStatus.ACTIVE,
+        },
+      });
+    if (!stackResult) return [null, 'Stack Id is not corrected'];
+    const currentTime = new Date();
+    if (roomResult.openTime > currentTime)
```
Comment (CuonggTV) [inline src/rooms/rooms.service.ts]: You can this before finding stack for higher performance

File: src/rooms/rooms.service.ts [hunk 6]
```diff
@@ -222,4 +240,89 @@ export class RoomsService {
...
+      return [null, 'Room is started, cant update any more'];
+    const currentStack = roomResult.questionStack;
+    currentStack.status = QuestionStackStatus.ACTIVE;
```
Comment (CuonggTV) [inline src/rooms/rooms.service.ts:255]: Sever crashed for undefined
image

**Review by CuonggTV — APPROVED** [review]

**Pull request merged**